### ISSUE TITLE
Classname suffixes vanish when building in release mode

### ISSUE THREAD

**Issue by psiclopz** [JedWatson/react-select#4151]
I'm trying to write Selenium tests for the latest version of React-Select.  When working in webpack dev mode the select has divs with stuff like 
css-[hash]-ValueContainer
css-[hash]-IndicatorsContainer
css-[hash]-IndicatorSeparator

which is great for targeting xpaths.  But then I deploy (which uses the release mode) and all of the nice class names vanish.  So no more -ValueContainer, -IndicatorsContainer, -IndicatorSeparator, etc...

I can only assume that this is happening to aid with minimization, but it's really frustrating.  Is there a way to force those class name suffixes to persist in non-dev env?

**Comment by Rall3n** [comment]
I suggest using the props `className` and `classNamePrefix`.

```jsx
/* Sets a className attribute on the outer component */
className?: string,

/*
  If provided, all inner components will be given a prefixed className attribute.
  This is useful when styling via CSS classes instead of the Styles API approach.
*/
classNamePrefix?: string | null,
```

**Comment by psiclopz** [comment]
I could, there are other workarounds for xPaths as well.  But it seems buggy that it renders different html based on the environment.

**Comment by ebonow** [comment]
@psiclopz the behavior you are describing is not a bug, but rather how Emotion applies styling to various elements and it's not intended to ever be considered a reliable method of querying DOM elements as those classNames are dynamically generated. It's highly recommended you do not target those classNames.

Instead you can follow the advice of @Rall3n if you require the use of classNames, by applying a classNamePrefix and/or className.

For a complete example of the different classNames generated by using classNamePrefix, please refer to [this thread](https://github.com/JedWatson/react-select/issues/3671). It should be very helpful and considered the correct approach if you plan on using css stylesheets.

For more guidance on styling react-select, you can visit [here: ](https://react-select.com/styles)

**Comment by NathanQ** [comment]
There are 2 inputs in the react-select component. One is hidden and one to interact with. The one to add a value to starts with `react-select`. This is how I got selenium to find it. 
```js
// component
<AsyncSelect
    id='specific-id'
    ...
/>
```

```py
# selenium test
input = browser.find_element(
    by=By.XPATH,
    value="//div[@id='specific-id']//input[starts-with(@id, 'react-select')]"
)
```
